src/context.cc

```diff
@@ -1373,48 +1373,45 @@ error Context::downloadUpdate() {
             return err;
         }
 
-        // Get update check URL
-        std::string update_url("");
-        err = updateURL(&update_url);
-        if (err != noError) {
-            return err;
+        if (HTTPSClient::Config.AppVersion.empty()) {
+            return error("Cannot check for updates without app version");
         }
 
         // Ask Toggl server if we have updates
         std::string url("");
         std::string version_number("");
         {
             HTTPSRequest req;
-            req.host = urls::API();
-            req.relative_url = update_url;
+            req.host = "https://toggl.github.io";
+            req.relative_url = "/toggldesktop/assets/releases/updates.json";
 
             TogglClient client;
             HTTPSResponse resp = client.Get(req);
             if (resp.err != noError) {
                 return resp.err;
             }
 
-            if ("null" == resp.body) {
-                logger().debug("The app is up to date");
-                if (UI()->CanDisplayUpdate()) {
-                    UI()->DisplayUpdate("");
-                }
-                return noError;
-            }
-
             Json::Value root;
             Json::Reader reader;
             if (!reader.parse(resp.body, root)) {
                 return error("Error parsing update check response body");
             }
+            auto latestVersion = root[shortOSName()][update_channel];
```

Comment:
What if the json format changes in the future or is generated wrong? I think we should be more defensive in this case.

Reply:
It won't fail as `Json::nullValue` when indexed will return `Json::nullValue`.
I can think of throwing an error when we can't find any version in the end, but that also means we would have an error when testing a new platform on a not-yet-released channel, which is okay I guess.